src/1215_a_cpp_approach_to_physical_units.bs

```diff
@@ -171,12 +171,28 @@ std::cout << "Average speed: "
           << std::units::quantity_cast<std::units::kilometre_per_hour>(speed) << '\n';
 ```
 
+Errors in variable dimensions shall be made clear at compile time, rather than having a chance
+to fail at runtime. For example:
+
+```cpp
+constexpr std::units::percent lightspeed_limit(std::units::Length auto d, std::units::Time auto t)
+{
+    return avg_speed(t, d) / std::units::constants::speed_of_light; 
+}
+```
+Should fail to compile with:
+```text
+example.cpp: in function 'constexpr std::units::Velocity auto avg_speed(std::units::Length auto d, std::units::Time auto t)'
+example.cpp: error: t is not a Length.
+example.cpp: error: d is not a Time.
+```
+
```

Comment:
Hmm, I am not sure about the `lightspeed_limit()` example. It is not clear what a `std::units::percent` is. Also, it mentions `std::units::constants::speed_of_light` constant. Let's not scare people as of yet and not start the discussion if `std::units::constants` is the right place to have such values. There will be another paper with the description of the library after this one. Also, it mentions `example.cpp: error: t is not a Length.`. But why the error happen (no invalid input presented in the paper). Another point is that none compiler on the market is able to produce such an error as of now.

Reply:
Hmm... Maybe you are right about skipping this one.

The `example.cpp: error: t is not a Length.` message is right.  I switched the arguments with the signature of the example above: `avg_speed(Length, Time)`

If you would still like an example, I can think about another one.

Reply:
I think that this example is not needed. ISO committee members are already familiar with Concepts so for them, this case just duplicates the first example as we do not have to explicitly say to them again that wrong types will fail with a nices error logs ;-) This subject is covered later on anyway too.